include/StringUtils.h

```diff
@@ -0,0 +1,15 @@
+#pragma once
+
+#include <string>
+#include <vector>
+
+namespace StringUtils
```

Comment:
The functions here should all return a new string (or vector) that contains the result of the operation (no inplace modifications).

Ex.: ltrim should be `std::string ltrim(const std::string&)`.

Also, the functions should be renamed to something that fits the Framework's naming convention better:
- `replace_all` -> `ReplaceAll`
- `erase_all` -> `EraseAll`
- `split` -> `Split`
- `ltrim` -> `TrimStart`
- `rtrim` -> `TrimEnd`
- `trim` -> `Trim`

Besides that, the implementation of these functions are pretty good.